### PR TITLE
16624 - Auth Web - Change NR/business action buttons

### PR DESCRIPTION
**Pull request by chenhongjing**
*Issue #:*
https://github.com/bcgov/entity/issues/16624

*Description of changes:*
~- Added feature flag `enable-new-actions-menu`~
- Changed words for primary actions
- Changed words for secondary actions
- Added open externally icon as needed
- Added new icons according to uxpin
- Moved "Columns to Show" to table header bar
- Added hover texts for some actions
- Added Business Unaffiliated Modals
   show info dialog when trying to open dashboard of unaffiliated business
- Updated unit tests
- App version = 2.2.7

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

**Comment by seeker25** [comment]
Needs a resync, does this need to go in for Tuesday's demo?

File: auth-web/src/components/auth/manage-business/AccountAuthorizationRequest.vue
Comment (seeker25) [inline auth-web/src/components/auth/manage-business/AccountAuthorizationRequest.vue:45]: Thank you for this

**Comment by seeker25** [comment]
Is it possible we could include the resync in a seperate PR? That way it's clear between your changes and what was previously done?

**Comment by severinbeauvais** [comment]
> Does this need to go in for Tuesday's demo?

Entities Team does not have this feature as a sprint goal and wasn't planning to demo it... but we could. What do you think -- should we demo this from the feature branch? Or is the plan to demo all the changes at once, later?

cc: @jdyck-fw

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:642]: This declaration will override the variable in the parent scope, but this assignment probably won't be visible in line 644.

Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:642]: This is the cause of one code smell (in sonarcloud). I resolved the other one as "won't fix".

File: auth-web/devops/vaults.env
Comment (severinbeauvais) [inline auth-web/devops/vaults.env:14]: This file is the list of 1Password keys that need to be made available as environment variables.

Please also pre-populate the .env.sample file with these keys (and values if they are not secret).

File: auth-web/src/components/auth/common/ModalDialog.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/common/ModalDialog.vue:101]: Please show a screenshot with the effect of this change.

File: auth-web/src/components/auth/manage-business/EntityManagement.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/EntityManagement.vue:591]: Please show screenshots of the 2 variations of this new dialog.

File: auth-web/src/models/affiliation.ts
Comment (severinbeauvais) [inline auth-web/src/models/affiliation.ts:73]: These are the new fields coming from Namex, right?

File: auth-web/src/util/config-helper.ts
Comment (severinbeauvais) [inline auth-web/src/util/config-helper.ts:135]: Was it decided that this URL is "close enough", or do you need a new key with the fragment to the forms?

File: auth-web/src/util/constants.ts
Comment (severinbeauvais) [inline auth-web/src/util/constants.ts:174]: Good comments but the codes will become "modernized" via a FF, so I recommend not having these comments after all.

File: auth-web/src/util/constants.ts
Comment (severinbeauvais) [inline auth-web/src/util/constants.ts:200]: This list is a duplicate of https://github.com/bcgov/bcrs-shared-components/blob/main/src/modules/corp-type-module/corp-type-module.ts and it would be best to not have both.

I realize changing over to using the shared enum will take some work. I'm thinking we accept the change above and create a tech debt ticket to fix this later. @seeker25 , do you have a strong preference for changing this now instead?

File: auth-web/tests/unit/components/AffiliatedEntityTable.spec.ts
Comment (severinbeauvais) [inline auth-web/tests/unit/components/AffiliatedEntityTable.spec.ts:96]: What do you think about mocking the sessionStorage call instead? I'm OK either way.

File: auth-web/src/components/auth/manage-business/EntityManagement.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/EntityManagement.vue:263]: Is this class (and dialog-class on line 267) used for anything?

(I saw you use the ref in the unit tests.)

**Comment by severinbeauvais** [comment]
Please run the GCB after fixing the code conflicts. (That's probably why that check failed.)

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:21]: I like to see an explicit function call here, eg, `@change="updateHeaders($event)"`.

Also, why not call `getHeaders` directly here?

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:22]: 1. do you need "hide-details" (like the old one) to make it less tall?
2. please indent props under the main element tag
3. since you don't need the double-binding on `selectedColumn` (v-model), I suggest changing it to a `value` prop instead (and maybe call it initialColumns or something)

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:125]: There's an indent issue here.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:126]: Does this get displayed for all primary actions?

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:179]: Please comment in the ticket the scenarios that you now support (including this one) -- for QA.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:625]: This should check a FF also.

There is a FF for this in LD "entity-ui" project but THIS app uses a different LD project. I really don't want to duplicate this flag in "sbc-auth" project. @seeker25 Any ideas?

image

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:652]: Did you confirm with UX team that BC/CC/ULC/CP/BEN show the first label and the others show the second label?

I thought the logic was "if allow restore then show button else show something else"...?

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:593]: It might be better to refactor the logic so that "COLIN" is the last choice in goToRegister().

(Right now, you don't check "isOtherEntities" in goToRegister().)

I would ask Janis or Mihai which is the better last/fallback choice: COLIN or forms.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:734]: Per comment above, refactor here?

```
if (modernized)
  then ...
else if (societies)
  then ...
else if (other)
  then ...
else
  goToCorpOnline()
```

**Review by severinbeauvais — COMMENTED** [review]
I like the logic and breakdown of functions.

I wonder if it would better to create a sub-component for the actions, since you're adding a ton of code here that isn't strictly related to the "affiliated entity table" itself?

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:841]: Wrong flag?

Comment (chenhongjing) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:841]: :scream:

File: auth-web/src/util/constants.ts
Comment (seeker25) [reply auth-web/src/util/constants.ts:200]: Rather not block - we can deal with it later

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (seeker25) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:624]: We could add a label for auth-web - It looks like it's possible to have two client instances.  Would probably make double the usage though?

**Comment by seeker25** [comment]
```
      const isAddFormValid = (
Remove this useless assignment to variable "isAddFormValid".
        !!businessIdentifier.value &&
        !!passcode.value &&
        (!isFirm.value || (isCertified.value && !!certifiedBy.value)) &&
        addBusinessForm.value.validate()
      )
      ```

can remove -^

File: auth-web/src/util/constants.ts
Comment (severinbeauvais) [reply auth-web/src/util/constants.ts:200]: Ok with me. Hongjing, please create a tech debt ticket for this.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:624]: The label isn't used externally, I think.

Two client instances would be nice, but yeah, it might count twice towards usage.

I wonder if we should move to a single LaunchDarkly project for sbc-auth + entity-ui?

**Comment by seeker25** [comment]
I think this can go into `main`?

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2422-czkxdizn.web.app

**Comment by severinbeauvais** [comment]
> What's the point of feature flagging then? @severinbeauvais - sounds like double the testing and more cleanup to do.

You're right -- we don't need both. @chenhongjing do you have a preference whether to merge to feature branch and remove the FF, or merge to main branch and keep the FF?

I think we should merge to the feature branch because there's also a change to the "columns to show" that isn't feature flagged, and this PR has alredy been rebased here so it's less work.

**Comment by seeker25** [comment]
change to the "columns to show", probably similar to the changes I'm working on?

before
image

after
image

**Comment by severinbeauvais** [comment]
> change to the "columns to show", probably similar to the changes I'm working on?

Yup, same thing. Was it a todo item in your ticket? It was in 16624.

File: auth-web/src/components/auth/common/ModalDialog.vue
Comment (chenhongjing) [reply auth-web/src/components/auth/common/ModalDialog.vue:101]: It's to add a new css style for dialog. It's used for unaffiliated modals only (refers to uxpin https://preview.uxpin.com/a86dccdd20ac62828f965c5eea10f81b828a9853#/pages/163299967/simulate/sitemap).

File: auth-web/src/components/auth/manage-business/EntityManagement.vue
Comment (chenhongjing) [reply auth-web/src/components/auth/manage-business/EntityManagement.vue:597]: #### The new dialog: 
image

The wording is slightly different according to different NR types

File: auth-web/src/models/affiliation.ts
Comment (chenhongjing) [reply auth-web/src/models/affiliation.ts:91]: Yes 😃

File: auth-web/src/util/config-helper.ts
Comment (chenhongjing) [reply auth-web/src/util/config-helper.ts:135]: Confirmed with Janis. It's fine. No need to add a new one.

File: auth-web/src/components/auth/manage-business/EntityManagement.vue
Comment (chenhongjing) [reply auth-web/src/components/auth/manage-business/EntityManagement.vue:264]: > It's to add a new css style for dialog. It's used for unaffiliated modals only (refers to uxpin https://preview.uxpin.com/a86dccdd20ac62828f965c5eea10f81b828a9853#/pages/163299967/simulate/sitemap).

> #### The new dialog:
> image
> 
> The wording is slightly different according to different NR types

It's for unaffiliated modals. Please see comments and screenshot above.

File: auth-web/src/components/auth/common/ModalDialog.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/common/ModalDialog.vue:101]: OK. Can you show a screenshot of it from your branch?

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (chenhongjing) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:126]: No, it only shows for those with externally open icons, except for `Download Form`.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (chenhongjing) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:646]: Yes, confirmed with Janis.

Screenshots from RC for reference:
image
image

I'll think of another name for this function. It needs better naming. :point_right: `isAllow...` is not accurate.

Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:646]: Renaming the function is a good idea.

Does this action open the business' dashboard (Filings UI)?

Do we need a feature flag to enable/disable the action (else redirect to COLIN), for businesses we currently don't allow to restore/reinstate?

File: auth-web/src/components/auth/common/ModalDialog.vue
Comment (chenhongjing) [reply auth-web/src/components/auth/common/ModalDialog.vue:101]: Same as the below comments 😃 It keeps action button centered while the title and texts align left.
image

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:6]: Please indent element properties.

File: auth-web/src/util/constants.ts
Comment (severinbeauvais) [reply auth-web/src/util/constants.ts:200]: Hongjing, I'll create the new ticket, so you can focus on the other comments.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (chenhongjing) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:646]: > Do we need a feature flag to enable/disable the action (else redirect to COLIN), for businesses we currently don't allow to restore/reinstate?

I'm thinking of this. Will restore/reinstate be implemented before this branch merges into main? If so, I think no need to add a new FF. Now for button `Restore Now` and `Reinstate Now`, it will go to business dashboard if the business of NR is affiliated. Otherwise, it displays `Business unavailable` dialog.

File: auth-web/src/util/constants.ts
Comment (chenhongjing) [reply auth-web/src/util/constants.ts:200]: Wait... `CorpTypes` has 3 overloaded values (TMP, NR, RTMP) which `CorpTypeCd` doesn't have. Is it good to add them to `CorpTypeCd`?

**Comment by chenhongjing** [comment]
> ```
>       const isAddFormValid = (
> Remove this useless assignment to variable "isAddFormValid".
>         !!businessIdentifier.value &&
>         !!passcode.value &&
>         (!isFirm.value || (isCertified.value && !!certifiedBy.value)) &&
>         addBusinessForm.value.validate()
>       )
> ```
> 
> can remove -^

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue]: I created a new FF (in LD sbc-auth project), `supported-restoration-entities`, that you can use here to determine whether the subject entity type can be restored in LEAR or not.

In Dev and Test, this flag should return "BC BEN CC ULC", and in Prod, "".

**Review by seeker25 — APPROVED** [review]
LGTM

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue]: Yeah, so this is a duplicate of the flag in entity-ui, but it's a temporary flag so hopefully we remember to remove both in the future when restorations is fully implemented.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue]: This conversation is resolved by a change to `goToRegister` below.

Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue]: I still think this isn't the right way to check for a "COLIN" entity. 

I think (but please confirm if this makes sense), that if the entity is not modernized (according to FF), and not a Society, and not an Other, then it's a COLIN. See comment below re: goToRegister().

**Comment by chenhongjing** [comment]
> > What's the point of feature flagging then? @severinbeauvais - sounds like double the testing and more cleanup to do.
> 
> You're right -- we don't need both. @chenhongjing do you have a preference whether to merge to feature branch and remove the FF, or merge to main branch and keep the FF?
> 
> I think we should merge to the feature branch because there's also a change to the "columns to show" that isn't feature flagged, and this PR has alredy been rebased here so it's less work.

No need for FF `enable-new-actions-menu`, right?

File: auth-web/tests/unit/components/AffiliatedEntityTable.spec.ts
Comment (severinbeauvais) [reply auth-web/tests/unit/components/AffiliatedEntityTable.spec.ts:116]: This is the way other tests do it, so leave it this way. Resolving.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue]: We have implemented restoration filings for some entity types only, but not fully released it to Prod. So, yes, a FF is needed here.

Maybe confirm with Janis or Mihai what should happen if there is a REH NR for an entity type not included in the FF.

**Comment by severinbeauvais** [comment]
> > > What's the point of feature flagging then? @severinbeauvais - sounds like double the testing and more cleanup to do.
> > 
> > 
> > You're right -- we don't need both. @chenhongjing do you have a preference whether to merge to feature branch and remove the FF, or merge to main branch and keep the FF?
> > I think we should merge to the feature branch because there's also a change to the "columns to show" that isn't feature flagged, and this PR has alredy been rebased here so it's less work.
> 
> No need for FF `enable-new-actions-menu`, right?

Correct. I will delete it in LaunchDarkly.

File: auth-web/devops/vaults.env
Comment (severinbeauvais) [inline auth-web/devops/vaults.env:14]: @pwei1018 It looks like these new keys are not being populated in the temp URL (https://bcregistry-account-dev--pr-2422-czkxdizn.web.app/).

Comment (severinbeauvais) [reply auth-web/devops/vaults.env:14]: Never mind - I figured it out.

@chenhongjing , you need to add these 2 new keys to devops/cloudbuild-pr.yaml, same as Karim did here: https://github.com/bcgov/business-create-ui/pull/559/files#r1287700270

Comment (severinbeauvais) [reply auth-web/devops/vaults.env:14]: And then, run the Google Cloud Build again.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue]: This was done in a recent commit. Thanks! Resolving.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (chenhongjing) [inline auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue:851]: This is based on another uxpin I got from Janis. I also add it in the ticket.
https://preview.uxpin.com/a86dccdd20ac62828f965c5eea10f81b828a9853#/pages/163836595/simulate/sitemap

File: auth-web/src/util/constants.ts
Comment (severinbeauvais) [reply auth-web/src/util/constants.ts:200]: No, but that's a problem for the other ticket :)

(It might be possible to extend an enum or something. Or fix the code to not use overloaded values...)

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2422-czkxdizn.web.app

**Comment by severinbeauvais** [comment]
This looks great!

image

Should there be a tooltip for the Download Forms action above it?

**Comment by chenhongjing** [comment]
> This looks great!
> 
> image
> 
> Should there be a tooltip for the Download Forms action above it?

No, no tooltip for Download Form.

File: auth-web/devops/vaults.env
Comment (chenhongjing) [reply auth-web/devops/vaults.env:14]: Did the changes, but I find it can't go to form page. Why does the problem still exist?

Comment (severinbeauvais) [reply auth-web/devops/vaults.env:14]: Patrick needs to configure something for us so that the 1Password keys are correctly converted into env variables and then built into the app. I've asked him to do that just now. This should be fixed by tomorrow morning.

File: auth-web/devops/cloudbuild-pr.yaml
Comment (pwei1018) [inline auth-web/devops/cloudbuild-pr.yaml:24]: If the environment variables is not the secret or password, don't need to set it in this file.

Comment (severinbeauvais) [reply auth-web/devops/cloudbuild-pr.yaml:24]: Where do we set it then?

PS These values are already in 1Password. This is to fetch the values from there, right?

File: auth-web/devops/cloudbuild-pr.yaml
Comment (pwei1018) [inline auth-web/devops/cloudbuild-pr.yaml:26]: If the environment variables is not the secret or password, don't need to set it in this file.

**Comment by seeker25** [comment]
Could we get this merged today? There are quite a few people waiting for this to get done so I can merge my PR.

File: auth-web/devops/cloudbuild-pr.yaml
Comment (severinbeauvais) [reply auth-web/devops/cloudbuild-pr.yaml:26]: As with the Corp Forms URL above, we can get this from 1Password, so shouldn't it be here?

Could I request from you a demo of how env keys are populated now, maybe at the Developer Sprint Demo meeting today? I think I don't understand this very well :)

**Comment by severinbeauvais** [comment]
> Could we get this merged today? There are quite a few people waiting for this to get done so I can merge my PR.
> 
> Otherwise I'm going to merge my PR, and you'll have to rebase / go through merge conflicts again.
> 
> Tay, Rodrigo, Jia, Hrvoje are all waiting

Yeah, let's do it ASAP.

Hongjing, can this PR me merged as-is? I think  you have a few more small comments/changes but they can go in another PR.

**Review by severinbeauvais — APPROVED** [review]
Approved so we can merge this ASAP, with the understanding that any outstanding comments will be handled in a future PR.

File: auth-web/src/util/constants.ts
Comment (severinbeauvais) [reply auth-web/src/util/constants.ts:174]: Resolved in #2433 .

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue]: Now using FF (in #2433 ). Resolving.

File: auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue
Comment (severinbeauvais) [reply auth-web/src/components/auth/manage-business/AffiliatedEntityTable.vue]: Resolved in #2433 .

**Pull request merged**
